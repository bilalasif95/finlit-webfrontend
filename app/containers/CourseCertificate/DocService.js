import { savePDF } from '@progress/kendo-react-pdf';

class DocService {
  createPdf = html => {
    savePDF(html, {
      paperSize: 'A4',
      fileName: 'FinLitCertificate.pdf',
      margin: 0,
      fontFamily: 'Lato',
    });
  };
}

const Doc = new DocService();
export default Doc;
