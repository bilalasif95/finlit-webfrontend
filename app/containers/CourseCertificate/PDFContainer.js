import React from 'react';
import PropTypes from 'prop-types';

export default function PDFContainer(props) {
  const bodyRef = React.createRef();
  const createPdf = () => props.createPdf(bodyRef.current);
  return (
    <div className="pdf-container">
      <div className="pdf-toolbar">
        <button type="button" onClick={createPdf}>
          Generate Certificate as PDF
        </button>
      </div>
      <div className="pdf-body" ref={bodyRef}>
        {props.children}
      </div>
    </div>
  );
}

PDFContainer.propTypes = {
  createPdf: PropTypes.any,
  children: PropTypes.any,
};
