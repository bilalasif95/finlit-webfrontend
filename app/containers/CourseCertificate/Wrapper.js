import styled from 'styled-components';

const Wrapper = styled.div`
  background-color: #fff;
  padding: 2rem 0 0;
  .course_certificate {
    border: 1px solid #e9e9e9;
    // transform: rotate(90deg);
    position: relative;
    max-width: 80rem;
    .circle_bg {
      bottom: 0;
      display: none;
      position: absolute;
      right: 0;
      width: 25rem;
    }
    .header {
      // background-color: #255f2b;
      background: url('https://i.imgur.com/4mnVfVZ.png') no-repeat;
      background-size: cover;
      background-position: top center;
      border-bottom: 10px solid #ffd430;
      padding: 4.5rem 6rem;
      z-index: 1;
      h4 {
        color: #fff;
        font-size: 2.5rem;
        font-weight: 300;
        line-height: 24px;
        margin: 0 0 2.5rem;
      }
      h2 {
        color: #fff;
        font-size: 3.5rem;
        font-weight: 500;
        line-height: 24px;
        margin: 0 0 0.25rem;
      }
    }
  }
  .certificate_details {
    padding: 4rem 6rem;
    .name_date {
      display: flex;
      flex-wrap: wrap;
      justify-content: flex-start;
      width: 100%;
      h1 {
        color: #3e952e;
        font-size: 4rem;
        font-style: italic;
        line-height: 44px;
        margin: 0 0 1rem;
      }
      .comp_date {
        align-items: center;
        display: inherit;
        justify-content: space-between;
        width: 100%;
        p {
          color: #404041;
          font-size: 1.15rem;
          margin: 0;
        }
      }
    }
    .course_title {
      color: #404041;
      font-size: 1.15rem;
      margin: 2rem 0 5rem;
    }
    .certified_by {
      width: 100%;
      img {
        margin: 0 0 0.5rem;
        width: 10rem;
      }
      p {
        color: #404041;
        font-size: 1rem;
        font-style: italic;
        margin: 0;
      }
    }
  }
  .footer {
    background-color: #41952e;
    padding: 1.5rem 6rem;
    position: relative;
    a {
      color: #fff;
      font-size: 1.25rem;
      margin: 1.5rem 0 0;
    }
    .logo {
      background-color: #255f2b;
      bottom: 1.5rem;
      padding: 2.5rem 10rem 2.5rem 2.5rem;
      position: absolute;
      right: 0;
      img {
        width: 8rem;
      }
    }
  }
`;

export default Wrapper;
