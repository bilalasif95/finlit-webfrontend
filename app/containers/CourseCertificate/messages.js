/*
 * Course Certificate Messages
 *
 * This contains all the text for the Course Certificate.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.containers.CourseCertificatePage';

export default defineMessages({
  FinLitAcademy: {
    id: `${scope}.FinLitAcademy`,
    defaultMessage: 'FinLit Professional Academy',
  },
  CertificateCompletion: {
    id: `${scope}.CertificateCompletion`,
    defaultMessage: `Certificate of Completion`,
  },
  ChangePassword: {
    id: `${scope}.ChangePassword`,
    defaultMessage: 'Change Password',
  },
  Desc: {
    id: `${scope}.Desc`,
    defaultMessage: `Aenean sed lorem est. Sed quis neque ut nibh suscipit
    imperdiet ac non augue. Aenean ornare sit amet lectus
    non tristique. Nunc ut volutpat lectus. Nulla 
    velit augue, pulvinar sed nisi sit amet, eleifend fermentum est.`,
  },
});
