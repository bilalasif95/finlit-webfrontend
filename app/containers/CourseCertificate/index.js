/*
 * Course Certificate Page
 *
 */
import React from 'react';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
// import { withStyles } from '@material-ui/core/styles';
import { Container } from 'reactstrap';
// import Img from '../../components/Img';
import messages from './messages';
import Wrapper from './Wrapper';
import Logo from '../../images/logowhite.svg';
// import circlebg from '../../images/circlebg.png';
import Doc from './DocService';
import PdfContainer from './PDFContainer';

export default function CourseCertificate() {
  const createPdf = html => Doc.createPdf(html);
  return (
    <div className="sub_pages">
      <Helmet>
        <title>Course Certificate</title>
        <meta name="description" content="Course  Certificate Page" />
      </Helmet>
      <Wrapper>
        <Container fluid="xl">
          <PdfContainer createPdf={createPdf} paperSize="A4">
            <div className="course_certificate">
              {/* <Img src={circlebg} alt="bg" height="100%" width="100%" className="circle_bg" /> */}
              <div className="header">
                <h4>
                  <FormattedMessage {...messages.FinLitAcademy} />
                </h4>
                <h2>
                  <FormattedMessage {...messages.CertificateCompletion} />
                </h2>
              </div>
              <div className="certificate_details">
                <div className="name_date">
                  <h1>Chad Sanders</h1>
                  <div className="comp_date">
                    <p>Has successfully completed.</p>
                    <p>Date of Achievement: 14-Feb-2021</p>
                  </div>
                  <div className="course_title">
                    <p>Course</p>
                    <p>
                      <FormattedMessage {...messages.CourseName} />
                    </p>
                  </div>
                  <div className="certified_by">
                    <p className="name">RandyHayes</p>
                    <p>Randy Hayes</p>
                    <p>General Manager</p>
                    <p>FinLit Learning Experiences Group</p>
                  </div>
                </div>
              </div>
              <div className="footer">
                <Link to="/">www.finlit.com</Link>
                <div className="logo">
                  <img src={Logo} alt="FinLit" height="100%" width="100%" />
                </div>
              </div>
            </div>
          </PdfContainer>
        </Container>
      </Wrapper>
    </div>
  );
}
