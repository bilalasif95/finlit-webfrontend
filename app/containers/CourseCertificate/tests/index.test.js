import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import CourseCertificate from '../index';

describe('<CourseCertificate />', () => {
  it('should render its heading', () => {
    const { container } = render(
      <IntlProvider locale="en">
        <CourseCertificate />
      </IntlProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
