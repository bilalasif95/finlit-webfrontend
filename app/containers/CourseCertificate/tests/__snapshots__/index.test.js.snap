// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseCertificate /> should render its heading 1`] = `
<div
  class="sub_pages"
>
  <div
    class="Wrapper-tcjrjc-0 UyYOk"
  >
    <div
      class="container-xl"
    >
      <div
        class="course_certificate"
      >
        <div
          class="header"
        >
          <h4>
            <span>
              FinLit Professional Academy
            </span>
          </h4>
          <h2>
            <span>
              Certificate of Completion
            </span>
          </h2>
        </div>
        <div
          class="certificate_details"
        >
          <div
            class="name_date"
          >
            <h1>
              Chad Sanders
            </h1>
            <div
              class="comp_date"
            >
              <p>
                Has successfully completed.
              </p>
              <p>
                Date of Achievement: 14-Feb-2021
              </p>
            </div>
            <div
              class="course_title"
            >
              <p>
                Course
              </p>
              <p>
                <span>
                  Complete Programming with Python : HandsOn Introduction for Beginners
                </span>
              </p>
            </div>
            <div
              class="certified_by"
            >
              <p
                class="name"
              >
                RandyHayes
              </p>
              <p>
                Randy Hayes
              </p>
              <p>
                General Manager
              </p>
              <p>
                FinLit Learning Experiences Group
              </p>
            </div>
          </div>
        </div>
        <div
          class="footer"
        >
          <a
            href="/"
          >
            www.finlit.com
          </a>
          <div
            class="logo"
          >
            <img
              alt="FinLit"
              height="100%"
              src="IMAGE_MOCK"
              width="100%"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
